nx-dev/feature-search/src/lib/algolia-search.tsx

```diff
@@ -0,0 +1,155 @@
+import { useState, useCallback, useRef, useEffect } from 'react';
+import { createPortal } from 'react-dom';
+import Link from 'next/link';
+import Head from 'next/head';
+import { useRouter } from 'next/router';
+import { DocSearchModal, useDocSearchKeyboardEvents } from '@docsearch/react';
+
+declare var document: any;
```

Comment:
We can make these types available by including the `dom` lib in `tsconfig.lib.json`.